effekt/shared/src/main/scala/effekt/machine/Transformer.scala

```diff
@@ -125,6 +125,20 @@ object Transformer {
         noteBlockParams(id, allParams)
         Def(Label(transform(id), allParams), transform(body), transform(rest))
 
+      case lifted.Def(id, tpe, block @ lifted.New(impl), rest) =>
+        // TODO freeParams?
+        // TODO deal with evidenve?
+        val interface = tpe.asInstanceOf[symbols.Interface]
+        val implTransformed = impl.clauses.sortBy[Int]({
+          case (operationName, _) =>
+            interface.ops.indexOf(operationName)
+        }).map({
+          case (_, lifted.BlockLit(params, body)) =>
+            // TODO we assume that there are no block params in methods
```

Comment:
This assumption may soon be invalidated.

Reply:
Yes. Will we have to change it at https://github.com/effekt-lang/effekt/pull/162/files#diff-2bc35700065c09eb82c800736c2625a2516ddb669bb20acbfce3d9af8dbca683R329, too?